src/main/java/rx/internal/operators/OperatorOnBackpressureLatest.java

```diff
@@ -0,0 +1,225 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package rx.internal.operators;
+
+import java.util.concurrent.atomic.*;
+
+import rx.Observable.Operator;
+import rx.*;
+
+/**
+ * An operator which drops all but the last received value in case the downstream
+ * doesn't request more.
+ */
+public final class OperatorOnBackpressureLatest<T> implements Operator<T, T> {
+    /** Holds a singleton instance initialized on class-loading. */
+    static final class Holder {
+        static final OperatorOnBackpressureLatest<Object> INSTANCE = new OperatorOnBackpressureLatest<Object>();
+    }
+    
+    /**
+     * Returns a singleton instance of the OnBackpressureLatest operator since it is stateless.
+     * @return the single instanceof OperatorOnBackpressureLatest
+     */
+    @SuppressWarnings("unchecked")
+    public static <T> OperatorOnBackpressureLatest<T> instance() {
+        return (OperatorOnBackpressureLatest<T>)Holder.INSTANCE;
+    }
+    
+    @Override
+    public Subscriber<? super T> call(Subscriber<? super T> child) {
+        final LatestEmitter<T> producer = new LatestEmitter<T>(child);
+        LatestSubscriber<T> parent = new LatestSubscriber<T>(producer);
+        producer.parent = parent;
+        child.add(parent);
+        child.add(producer);
+        child.setProducer(producer);
+        return parent;
+    }
+    /**
+     * A terminatable producer which emits the latest items on request.
+     * @param <T>
+     */
+    static final class LatestEmitter<T> extends AtomicLong implements Producer, Subscription, Observer<T> {
+        /** */
+        private static final long serialVersionUID = -1364393685005146274L;
+        final Subscriber<? super T> child;
+        LatestSubscriber<? super T> parent;
+        final AtomicReference<Object> value;
+        /** Written before done, read after done. */
+        Throwable terminal;
+        volatile boolean done;
+        /** Guarded by this. */
+        boolean emitting;
+        /** Guarded by this. */
+        boolean missed;
+        static final Object EMPTY = new Object();
+        static final long NOT_REQUESTED = Long.MIN_VALUE / 2;
+        public LatestEmitter(Subscriber<? super T> child) {
+            this.child = child;
+            this.value = new AtomicReference<Object>(EMPTY);
+            this.lazySet(NOT_REQUESTED); // not 
+        }
+        @Override
+        public void request(long n) {
+            if (n >= 0) {
```

Comment:
BackpressureUtils didn't exist at that point, so I am considering how to consolidate this type of logic as we keep repeating this type of non-trivial code and it is easy to get wrong. 

I'm okay with merging ... we really should spend some time figuring out the core patterns so we can encode the state machine, similar to what BackpressureUtils and AbstractOnSubscribe have started formalizing.